pkg/pipelines/imagerepo/imagerepo.go

```diff
@@ -0,0 +1,91 @@
+package imagerepo
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/namespaces"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+
+	"github.com/openshift/odo/pkg/pipelines/config"
+
+	res "github.com/openshift/odo/pkg/pipelines/resources"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ValidateImageRepo validates the input image repo.  It determines if it is
+// for internal registry and prepend internal registry hostname if neccessary.
+func ValidateImageRepo(imageRepo, registryURL string) (bool, string, error) {
+	components := strings.Split(imageRepo, "/")
+
+	// repo url has minimum of 2 components
+	if len(components) < 2 {
+		return false, "", imageRepoValidationErrors(imageRepo)
+	}
+
+	for _, v := range components {
+		// check for empty components
+		if strings.TrimSpace(v) == "" {
```

Comment:
This could be extracted to a function...

```go
func isBlank(s string) bool {
    return strings.TrimSpace(s) == "" || len(v) > len(strings.TrimSpace(s))
}
```

And then just

```go
if isBlank(v) {
    return imageRepoValidationErrors(imageRepo)
}
```

Reply:
Done.